pennylane/math/multi_dispatch.py

```diff
@@ -898,6 +880,37 @@ def detach(tensor, like=None):
     return tensor
 
 
+def jax_argnums_to_tape_trainable(qnode, argnums, expand_fn, args, kwargs):
+    """This functions gets the tape parameters from the QNode construction given some argnums (only for Jax).
+    The tape parameters are transformed to JVPTracer if they are from argnums. This function imitates the behavior
+    of Jax in order to mark trainable parameters.
+
+    Args:
+        qnode(qml.QNode): the quantum node.
+        argnums(int, list[int]): the parameters that we want to set as trainable (on the QNode level).
+        expand_fn(callable): the function that is expanding the tape.
+
+    Return:
+        list[float, jax.JVPTracer]: List of parameters where the trainable one are `JVPTracer`.
+    """
+    import jax
+
+    with jax.core.new_main(jax.ad.JVPTrace) as main:
+        trace = jax.ad.JVPTrace(main, 0)
+
+    args_jvp = [
+        jax.ad.JVPTracer(trace, arg, jax.numpy.zeros(1)) if i in argnums else arg

```

Comment:
> jax.numpy.zeros(1)

Does this assume scalar tape parameters?

Reply:
Yes I have changed it in the next PR https://github.com/PennyLaneAI/pennylane/pull/3847

Reply:
Great, thanks!